alf/algorithms/entropy_target_algorithm.py

```diff
@@ -0,0 +1,122 @@
+                 target_entropy=None,
+                 optimizer=None,
+                 debug_summaries=False):
+        """Create an EntropyTargetAlgorithm
+
+        Args:
+            action_spec (nested BoundedTensorSpec): representing the actions.
+            initial_log_alpha (float): initial value for log(alpha). The default
+                value corresponds to alpha ~ 0.05.
+            target_entropy (float): the lower bound of the entropy. If not
+                provided, a default value proportional to the action dimension
+                is used.
+            optimizer (tf.optimizers.Optimizer): The optimizer for training. If
+                not provided, will use the same optimizer of the parent
+                algorithm.
+            debug_summaries (bool): True if debug summaries should be created.
+        """
+        log_alpha = tfa_common.create_variable(
+            name='log_alpha',
+            initial_value=initial_log_alpha,
+            dtype=tf.float32,
+            trainable=True)
+        super().__init__(
+            optimizer=optimizer,
+            debug_summaries=debug_summaries,
+            name="EntropyTargetAlgorithm")
+
+        self._log_alpha = log_alpha
+        self._action_spec = action_spec
+
+        def _calc_default_target_entropy(spec):
+            dims = np.product(spec.shape.as_list())
+            if spec.dtype.is_floating:
+                e = -1
+            else:
+                min_prob = 0.01
+                p = min_prob
+                q = 1 - p
+                e = -p * np.log(p) - q * np.log(q)
+            return e * dims
+
+        if target_entropy is None:
+            flat_action_spec = tf.nest.flatten(self._action_spec)
+            target_entropy = np.sum(
+                list(map(_calc_default_target_entropy, flat_action_spec)))
+        self._target_entropy = target_entropy
+
+    def train_step(self, distribution):
+        """Train step.
+
+        Args:
+            distribution (nested Distribution): action distribution from the
+                policy.
+        Returns:
+            AlgorithmStep. `info` field is LossInfo, other fields are empty.
+        """
+        entropy, entropy_for_gradient = dist_utils.entropy_with_fallback(
+            distribution, self._action_spec)
+        alpha_loss = self._log_alpha * tf.stop_gradient(entropy -
+                                                        self._target_entropy)
+        alpha = tf.stop_gradient(tf.exp(self._log_alpha))
+        loss = alpha_loss
+        entropy_loss = -entropy
+        loss -= alpha * entropy_for_gradient
+
+        return AlgorithmStep(
+            outputs=(),
+            state=(),
+            info=LossInfo(
+                loss,
+                extra=EntropyTargetLossInfo(
+                    alpha_loss=alpha_loss, entropy_loss=entropy_loss)))
+
+    def calc_loss(self, training_info):
+        if self._debug_summaries and common.should_record_summaries():
```

Comment:
Is  condition `common.should_record_summaries()`  needed here as it already run under summary context with `run_under_record_context` ?

Reply:
It shouldn't affect the summary. But it may save a little computation for preparing the value to be summarized.